### ISSUE TITLE
Regarding accuracy and Tiny-yolo

### ISSUE THREAD

**Issue by sudharavali** [AlexeyAB/darknet#3219]
Hello @AlexeyAB,

Thank you for your great work ! 
I used tiny-yolo for object detection of my custom dataset for only one object. Now, I to say how accurate the model is, do I give it a test dataset and calculate the average of the bounding box scores or is the MAP value a better indicator. Can you please tell me what we should choose and why, I am a little confused. 
Also, do you think we can improve the accuracy if I used YOLO instead of Tiny-yolo ? 
Please let me know ! 

Thanks in advance.

**Comment by AlexeyAB** [comment]
@sudharavali Hi,

The mAP is all you need.

> Also, do you think we can improve the accuracy if I used YOLO instead of Tiny-yolo ?

Yes.

**Comment by sudharavali** [comment]
Thank you for your quick reply ! When I train on one dataset and want to check it's mAP score on another set of data, do I have to change any values in the obj.data file ?   

To elaborate, I am training a model to detect an object in a simulated environment and want to see how it will preform on the real samples using the mAP values. Please let me know how to do it. 

Thank you in advance !!

**Comment by AlexeyAB** [comment]
>  I just have one follow up question, When I train on **one dataset** and want to check it's mAP score on **another set of data**, do I have to change any values in the obj.data file ?

Just use in the `obj.data` files
```
train=one_dataset.txt
valid=another_dataset.txt
```


> To elaborate, I am training a model to detect an object in a simulated environment and want to see how it will preform on the real samples using the mAP values. Please let me know how to do it.

Jut use two datasets `one_dataset.txt` and `another_dataset.txt` as I described above.
Can you show examples of images from training and testing datasets?

**Comment by sudharavali** [comment]
Thanks alot for your prompt reply. 
Ofcourse I can show you the images :  

So for one detection model, the training images look like the following : 

![Drone_scene_1002](https://user-images.githubusercontent.com/33206551/58271783-dd02fd00-7d5a-11e9-9335-27563b685efc.jpg)

![Drone_scene_1446](https://user-images.githubusercontent.com/33206551/58271807-e724fb80-7d5a-11e9-8e9a-583b046fef6a.jpg)


For another detection model with another dataset, the training images look like the following : 

![B247](https://user-images.githubusercontent.com/33206551/58271714-afb64f00-7d5a-11e9-8932-48fcebecc81a.jpg)

![B355](https://user-images.githubusercontent.com/33206551/58271764-d2486800-7d5a-11e9-80c9-f1bf27de0a36.jpg)


I am trying to evaluate its accuracy and mAP on the images that look this way : 

![create100_original_312 jpg_fdd33f35-4969-4338-bd25-274e7609f1db](https://user-images.githubusercontent.com/33206551/58271609-6d8d0d80-7d5a-11e9-82af-3e206d16b0fb.jpg)

![create100_original_356 jpg_1f6c4835-8e63-48f9-bb92-6c8ce8f7e93f](https://user-images.githubusercontent.com/33206551/58271646-7ed61a00-7d5a-11e9-89da-5196810b021c.jpg)

![something](https://user-images.githubusercontent.com/33206551/58271668-8f869000-7d5a-11e9-818a-4fdaf68049dd.jpg)


I have one question pertaining to this : 
for : valid =another_dataset.txt , should it be 80 % of the dataset like the train dataset or 20% of the dataset like any other validation set ? 


Thank in advance !

**Comment by AlexeyAB** [comment]
`valid =another_dataset.txt` can be any dataset that doesn't intersect with training-dataset.

**Comment by sudharavali** [comment]
Thanks alot for your help ! 
After doing what you asked me to, these are my values observed for one detection model :

Validation Set : mAP score : 38 %, Bounding Box confidence score (average) : 49%
Test Set (on the real samples ) :  mAP score : 98.4 %, Bounding Box confidence score (average) : 4%

Can you tell me how to interpret this  ?
Can you tell why the bounding box confidence is so low when the mAP is so high for the test dataset. 

Thank you again in advance !

**Comment by AlexeyAB** [comment]
The confidence score is not important.

You need only for mAP.

> Can you tell why the bounding box confidence is so low when the mAP is so high for the test dataset.

What confidence threshold did you use to calculate confidence score?
By default =0.25, so average value can't be less than 25%.

**Comment by sudharavali** [comment]
> What confidence threshold did you use to calculate confidence score?

For 200 test images, the model didnt draw a bounding box on a majority of them. So I just did the sum of the individual confidence scores(many of which were of 0 value) / and divided it by total number of images (200 in this case) . That is how I calculated and got 4%.  

> By default =0.25, so average value can't be less than 25%

Can you tell me where to find this value?  Is this part of the config file  ?

**Comment by AlexeyAB** [comment]
` -thresh 0.25` flag: https://github.com/AlexeyAB/darknet#how-to-use-on-the-command-line

`./darknet detector test cfg/coco.data cfg/yolov3.cfg yolov3.weights -thresh 0.25`

**Comment by sudharavali** [comment]
I see that the threshold is set to 25% for images. But I don't get drone accuracy for few images. Please see the result attached below.

<img width="918" alt="Alexey" src="https://user-images.githubusercontent.com/33206551/58278496-2a3a9b00-7d6a-11e9-9633-5c05abbdf080.PNG">
